allennlp/nn/decoding/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""
+This module contains code for transition-based decoding.  If you want to do decoding for a
+vocabulary-based model, where the allowable outputs are the same at every timestep of decoding,
+this code is not what you are looking for, and it will be quite inefficient compared to other
+things you could do.
+
+This code assumes you have a transition-based model, where you start in some state, iteratively
```

Comment:
Nit: I'd move this to right after the first sentence.

Reply:
Done.